lib/ad_localize/runner.rb

```
@@ -7,29 +7,41 @@ def initialize
     end
 
     def run(args = ARGV)
-      LOGGER.log(:info, :green, "OPTIONS : #{options}")
-      input_files = (args + [options.dig(:drive_key)]).compact # drive_key can be nil
-      if input_files.length.zero?
-        LOGGER.log(:error, :red, "No CSV to parse. Use option -h to see how to use this script")
-      else
-        file_to_parse = args.first
-        LOGGER.log(:warn, :yellow, "Only one CSV can be treated - the priority goes to #{file_to_parse}") if input_files.length > 1
-        if args.empty?
-          options[:drive_file] = CsvFileManager.download_from_drive(options.dig(:drive_key), options.dig(:sheet_id), options.dig(:use_service_account))
-          file_to_parse = options.dig(:drive_file)
+        LOGGER.log(:info, :green, "OPTIONS : #{options}")
+        input_files = args
+        has_drive_key = options.dig(:drive_key)
+        raise 'No CSV to parse. Use option -h to see how to use this script' if input_files.length.zero? and !has_drive_key
+
+        files_to_parse = Array.new
+        if has_drive_key
+            LOGGER.log(:warn, :yellow, 'CSV file are ignored with the drive key option') if input_files.length > 1
+            options[:drive_file] = CsvFileManager.download_from_drive(options.dig(:drive_key), options.dig(:sheet_id), options.dig(:use_service_account))
+            files_to_parse.push(options.dig(:drive_file))
```

Can you raise an exception if options[:drive_file] is nil ?